public/app/features/alerting/unified/components/rule-list/util.ts

```diff
@@ -0,0 +1,71 @@
+import { addMilliseconds, formatDistanceToNow, formatDistanceToNowStrict, isBefore } from 'date-fns';
+
+import { dateTime, dateTimeFormat, isValidDate } from '@grafana/data';
+
+import { isNullDate, parsePrometheusDuration } from '../../utils/time';
+
+type NextEvaluation = {
+  humanized: string;
+  fullDate: string;
+};
+
+/**
+ * Best effort estimate for when the next evaluation will occur
+ * @TODO write a test for this
+ * @TODO move this somewhere else probably
+ */
+export function calculateNextEvaluationEstimate(
+  lastEvaluation?: string,
+  evaluationInterval?: string
+): NextEvaluation | undefined {
+  if (!lastEvaluation || !evaluationInterval) {
+    return;
+  }
+
+  if (!isValidDate(lastEvaluation)) {
+    return;
+  }
+
+  let intervalSize: number;
+  try {
+    intervalSize = parsePrometheusDuration(evaluationInterval);
+  } catch (error) {
+    return;
+  }
+
+  // paused alert rules will have their lastEvaluation set to a nil date
+  if (isNullDate(lastEvaluation)) {
+    return;
+  }
+
+  const lastEvaluationDate = Date.parse(lastEvaluation || '');
+  const nextEvaluationDate = addMilliseconds(lastEvaluationDate, intervalSize);
+
+  //when `nextEvaluationDate` is a past date it means lastEvaluation was more than one evaluation interval ago.
+  //in this case we use the interval value to show a more generic estimate.
+  //See https://github.com/grafana/grafana/issues/65125
+  const isPastDate = isBefore(nextEvaluationDate, new Date());
+  if (isPastDate) {
+    return {
+      humanized: `within ${formatDistanceToNow(nextEvaluationDate)}`,
```

Comment:
I think I'm not understanding this use case. I understood that in this case we should use the `interval value` instead of the `nextEvaluationDate`? 🤔

Reply:
I don't quite recall but there seems to be some additional context in https://github.com/grafana/grafana/issues/65125

Reply:
Ideally we'd receive the next evaluation timestamp from the server though :)